### ISSUE TITLE
unable to GROUP BY,  or ORDER BY on Android  -- "open /tmp/ql-tmp-260419420: no such file or directory"

### ISSUE THREAD

**Issue by rasabbey** [cznic/ql#132]
Hi Cznic, thank you very much for this library.

Please correct me if my conclusion is wrong.

After going through the source I believe this bug occurs because the "/tmp/" folder is not present or writable on a lot of Android devices.

As per the comment on line **122** in file  **github.com/cznic/file.go**
> TempFile provides a temporary file used for evaluating the GROUP BY, ORDER...

**Comment by cznic** [comment]
Please ping me in Two weeks, I am away from a PC.

**Comment by rasabbey** [comment]
No Problem, I put a temporary fix in my local copy but it is "crude".

Will ping in  a couple of weeks

**Comment by calmh** [comment]
FWIW, `TempFile` should work properly when built with `GOOS=android`.

**Comment by rasabbey** [comment]
@calmh , Thanks for pitching in.

Unfortunately am not able to build/compile with the flag `GOOS=android`

I am attempting to create a ".aar" library to be imported into Android Studio.

command used was: 
`GOOS=android gomobile bind -target=android -o myapp.aar -ldflags "-s -w"`

error received was:
```
gomobile: go install -p=4 -pkgdir=/golang/pkg/gomobile/pkg_android_arm -tags="" -ldflags -s -w gcflags=-shared -ldflags=-shared myapp failed: exit status 2
#myapp
/usr/lib/go-1.6/pkg/tool/linux_amd64/link: -shared and -buildmode=pie are incompatible
```

**Comment by cznic** [comment]
I have unfortunately never used Go with gomobile. Can you please try with latest stable, ie. go1.7, if, by chance, it gets any farther?

I am disappointed that FileTemp, as @calmh points out, does not adhere to platform specifics (if I understood the comment correctly).

**Comment by calmh** [comment]
No, it does, it's just that Android is not Linux, does not have a /tmp, and requires GOOS=android building. How to do that successfully I don't know, it's just that I ran into the same temp file issue on another project.

**Comment by cznic** [comment]
I'm back.

@rasabbey I'm not sure what I can do for you. Please let me know about the progress on this, if any.

**Comment by rasabbey** [comment]
@cznic welcome back, 

I only put a temporary fix to check if an sdcard exists and use that instead.
it is a crude fix, mainly because am working on other aspects of the application and
hoping to get back to the db section in about a week and do a better fix.

**Comment by cznic** [comment]
Looking forward your PR :smile:

**Comment by gedw99** [comment]
I also am using this with gomobile to make an aar package.

It's really because an a droid app has no ability to write anywhere.

Elias Naur will probably know the answer I suspect.
I am on mobile and can't find his github handle :)

**Comment by dc0d** [comment]
This is closed. But for some reason it is listed under open issues.

**Comment by cznic** [comment]
@dc0d This issue was never closed.

**Comment by dc0d** [comment]
Sorry and Thanks!
